### PR TITLE
Replaces the Rich Text Editor in the DataCollection with the Markdown Editor

### PR DESCRIPTION
**Pull request by FBartz28**
Replaces old Settings Form with new UI Form as well as implemented the Markdown Editor instead of the Rich Text Editor.

### CODE REVIEW

File: components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php [hunk 1]
```diff
@@ -31,7 +34,7 @@ class ilDclTableEditGUI
     protected ILIAS\HTTP\Services $http;
     protected ILIAS\Refinery\Factory $refinery;
     protected ilDclTableListGUI $parent_object;
-    protected int $obj_id;
```
Comment (iszmais) [inline components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php:34]: Why was this moved below the constructor?
If unintended please restore.

File: components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php [hunk 2]
```diff
@@ -93,6 +99,92 @@ public function create(): void
...
+            $start_date = $input_field->dateTime("Start")->withValue(null);
+            $end_date = $input_field->dateTime("End")->withValue(null);
+            $limited_action_period = $input_field->optionalGroup(["start" => $start_date, "end" => $end_date], "Limited Add / Edit / Delete Period")->withValue(null);
+        } else {
+            $start_date = $input_field->dateTime("Start")->withValue($this->table->getLimitStart());
+            $end_date = $input_field->dateTime("End")->withValue($this->table->getLimitEnd());
+            $limited_action_period = $input_field->optionalGroup(["start" => $start_date, "end" => $end_date], "Limited Add / Edit / Delete Period");
+        }
+
+        //😭
```
Comment (iszmais) [inline components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php]: Pls remove

File: components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php [hunk 3]
```diff
@@ -282,6 +375,7 @@ public function initForm(string $a_mode = "create"): void
 
     public function save(string $a_mode = "create"): void
     {
+        //TODO update method to load own form
```
Comment (iszmais) [inline components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php]: Pls remove trivial doc

File: components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php [hunk 4]
```diff
@@ -294,10 +388,13 @@ public function save(string $a_mode = "create"): void
         $ilTabs->activateTab("id_fields");
         $this->initForm($a_mode);
 
+        //        dd($this->checkInput($a_mode),$a_mode,$this->table_id);
```
Comment (iszmais) [inline components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php]: Pls remove trivial doc

File: components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php [hunk 5]
```diff
@@ -294,10 +388,13 @@ public function save(string $a_mode = "create"): void
...
+
         if ($this->checkInput($a_mode)) {
             if ($a_mode != "update") {
                 $this->table = ilDclCache::getTableCache();
             } elseif ($this->table_id) {
+                // we get here
```
Comment (iszmais) [inline components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php]: Pls remove trivial doc

File: components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php [hunk 6]
```diff
@@ -72,13 +76,15 @@ public function __construct(ilDclTableListGUI $a_parent_obj)
...
+    protected int $obj_id;
+
     public function executeCommand(): void
     {
         $cmd = $this->ctrl->getCmd();
 
         switch ($cmd) {
             case 'update':
-                $this->save("update");
+                $this->update();
```
Comment (iszmais) [inline components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php]: the case `update` can be removed since its identical to the default

**Review by iszmais — CHANGES_REQUESTED** [review]
Thank you for your PR!

Please apply the following changes:
- Pls use/overwrite the existing structure instead of implementing an additional
  - overwrite `save` instead of adding `update`
  - overwrite `initForm` instead of adding `initNewForm` 
  - adapt convetional workloads as in the old process (checkInput, permissionchecks etc)
- pls check all input types, some seem not to fit the saving (be aware the the new form might be a bit more type safe than the old one).
- Resolve all other threads

File: components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php (hunk 1 above)
Comment (FBartz28) [reply components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php:34]: I added it back and removed the duplicate from line 80 of the same variable

File: components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php (hunk 6 above)
Comment (FBartz28) [reply components/ILIAS/DataCollection/classes/Table/class.ilDclTableEditGUI.php]: Done!

**Comment by iszmais** [comment]
@FBartz28 Thank you for your contribution!

Due to the time frame for ILIAS 10 features i used your implementation as a state for a separate implementation: https://github.com/ILIAS-eLearning/ILIAS/pull/8253

Therefore, I will close this PR here.
Feel free to continue the discussion within the new PR.

Greetings,
@iszmais
